File: backend/src/main/java/com/tododuk/domain/user/repository/UserRepository.java
package com.tododuk.domain.user.repository;

import com.tododuk.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {
    Optional<User> findByApiKey(String apiKey);

    Optional<User> findByUserEmail(String email);
}

File: backend/src/main/java/com/tododuk/domain/user/repository/UserRepository.kt
package com.tododuk.domain.user.repository

import com.tododuk.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, Int> {
    fun findByApiKey(apiKey: String): Optional<User>
    fun findByUserEmail(email: String): Optional<User>
}
